opm/simulators/flow/OutputBlackoilModule.hpp

```diff
@@ -1115,6 +1115,10 @@ class OutputBlackOilModule : public GenericOutputBlackoilModule<GetPropType<Type
 
         if (!this->temperature_.empty())
             fs.setTemperature(this->temperature_[elemIdx]);
+        else {
+            const auto& fp = simulator_.vanguard().eclState().fieldProps();
+             fs.setTemperature(fp.get_double("TEMPI")[elemIdx]);
+        }
```

Comment:
I don't think this will work in general.  You're not supposed to have `TEMPI`-style information in restarted runs.

Reply:
Isnt TEMPI defaulted to a constant reservoir temperature.

Reply:
> Isn't `TEMPI` defaulted to a constant reservoir temperature.

Kind of (commit OPM/opm-common@3c9f56a14, current version in [`TableManager.cpp`](https://github.com/OPM/opm-common/blob/95795867abc0a2a9a8af1a568387fd4799b0de10/opm/input/eclipse/EclipseState/Tables/TableManager.cpp#L145-L147)), but it's usually not a good idea to depend on that.  You'll [just get](https://github.com/OPM/opm-common/blob/95795867abc0a2a9a8af1a568387fd4799b0de10/opm/input/eclipse/EclipseState/Grid/FieldProps.cpp#L1208-L1223) whatever temperature value happens to be the default in the [`RTEMP` keyword](https://github.com/OPM/opm-common/blob/95795867abc0a2a9a8af1a568387fd4799b0de10/opm/input/eclipse/share/keywords/000_Eclipse100/R/RTEMP#L12).  That temperature value may or may not be suitable for CO2STORE.